### PR TITLE
Dt search merge

### PR DESCRIPTION
**Pull request by DavidTTran**
Completed:
- Recipe index pictures/titles are links to their show page
- Search functionality moved to root path, only available after a user logs in
- Sad paths for search function
   - No provided/conflicting params generates 10 random recipes
   - Provides up to 10 recipes
- Merge conflicts

Refactor:
- Moved Spoonacular request/response into SpoonacularService
- Recipes are instantiated as a Recipe PORO
   - with `@id, @title, @image, @ingredients, @instructions, @time, @nutrition`

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -1,46 +1,24 @@
 class RecipesController < ApplicationController
   def index
-    params = format_params
-    connection = Faraday.new(url: 'https://api.spoonacular.com/',
-                             params: request_params)
-    request = connection.get('/recipes/complexSearch')
-    @recipes = JSON.parse(request.body, symbolize_names: true)
+    recipes = SpoonacularService.new(search_params).parse_recipes
+    @recipes = recipes[:results].map do |recipe_json|
```
Comment (mikez321) [inline app/controllers/recipes_controller.rb:4]: Probably a good place for later refactor

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -1,46 +1,24 @@
...
+      Recipe.new(recipe_json)
+    end
+    if @recipes.empty?
+      flash[:error] = "Sorry, we couldn't find any recipes matching your
+      specification. Here's some random recipes you might like."
+      redirect_to '/recipes'
+    end
```
Comment (mikez321) [inline app/controllers/recipes_controller.rb:11]: love the random recipe!

**Review by mikez321 — APPROVED** [review]
All the same stuff that was previously looking good is still looking good.  Nice work fixing the VCR stuff!

**Review by alex-latham — APPROVED** [review]
See closed PR for comment history.

**Pull request merged**